### PR TITLE
Improve `solc` compatibility

### PR DESCRIPTION
**Pull request by gnidan**
Re: #1 

- Use test from `truffle-contract` to test solc's compiled contract output
- ...

(WIP)

### CODE REVIEW

**Comment by gnidan** [comment]
@tcoulter is this the extent of the work for that ticket? the moved `truffle-contract` test now passes with minimal changes.

**Comment by gnidan** [comment]
Notes from @tcoulter:
> @gnidan I’m thinking you should just rip out all this garbage and do what needs to happen. If you spend time on all this, we’ll likely have to rip it out again.
>
> > my thinking is that formalizing a schema warrants high-level discussion / design
>
> We already have an implicit schema used by 1000's of people. My gut says formalizing it is the way to get more input, since it’s not obvious what the schema entails currently.
> 
> In the future where a json schema exists, I’d likely want something like this:
> 
> 1. A function called `validate`, which takes in a json blob, throws that to jsonschema and calls its validate function.
> 
> 2. `n` number of `fromX` functions, like `fromSolidityStandardOutput`, which are helpers that output a validated JSON blob based on specific input.
> 
> Aside: I’m sorry if this conflicts with anything above. My thoughts are evolving as we talk about this more. (edited)

(Since then, we've decided to go with json-schema as part of this effort after all)

File: index.js [hunk 1]
```diff
@@ -21,132 +142,82 @@ var TruffleSchema = {
...
-      existing_object = {};
-    }
+    Object.keys(existingObj.networks).forEach(function(network_id) {
+      obj.networks[network_id] = existingObj.networks[network_id];
+    });
 
-    obj.contract_name = obj.contract_name || "Contract";
+    // if (options.overwrite == true) {
+    //   e = {};
+    // }
```
Comment (gnidan) [inline index.js]: Oh I left this comment here - previously it was `existing_object = {}`, which wouldn't do anything, I would think except update the local variable to some other pointer. Is that overwrite behavior being relied on?

Comment (tcoulter) [reply index.js]: I'm not entirely sure why the original line is here. I'd take it out (like you did) and obliterate the comment. There might be repercussions but hopefully we can find them. My guess is it's cruft from an old version of the artifactor.

**Comment by gnidan** [comment]
Cleanup: merging this into `json-schema` to rework the corresponding PR #3 to include this (because more discussion is there)

**Pull request merged**